server/etcdserver/api/etcdhttp/health_test.go

```diff
@@ -0,0 +1,180 @@
+			alarms:           []*pb.AlarmMember{{MemberID: uint64(0), Alarm: pb.AlarmType_NOSPACE}},
+			healthCheckURL:   "/health",
+			expectStatusCode: http.StatusServiceUnavailable,
+			expectHealth:     "false",
+		},
+		{
+			name:             "Healthy if NOSPACE alarm is on and excluded",
+			alarms:           []*pb.AlarmMember{{MemberID: uint64(0), Alarm: pb.AlarmType_NOSPACE}},
+			healthCheckURL:   "/health?exclude=NOSPACE",
+			expectStatusCode: http.StatusOK,
+			expectHealth:     "true",
+		},
+		{
+			name:             "Healthy if NOSPACE alarm is excluded",
+			alarms:           []*pb.AlarmMember{},
+			healthCheckURL:   "/health?exclude=NOSPACE",
+			expectStatusCode: http.StatusOK,
+			expectHealth:     "true",
+		},
+		{
+			name:             "Healthy if multiple NOSPACE alarms are on and excluded",
+			alarms:           []*pb.AlarmMember{{MemberID: uint64(1), Alarm: pb.AlarmType_NOSPACE}, {MemberID: uint64(2), Alarm: pb.AlarmType_NOSPACE}, {MemberID: uint64(3), Alarm: pb.AlarmType_NOSPACE}},
+			healthCheckURL:   "/health?exclude=NOSPACE",
+			expectStatusCode: http.StatusOK,
+			expectHealth:     "true",
+		},
+		{
+			name:             "Unhealthy if NOSPACE alarms is excluded and CORRUPT is on",
+			alarms:           []*pb.AlarmMember{{MemberID: uint64(0), Alarm: pb.AlarmType_NOSPACE}, {MemberID: uint64(1), Alarm: pb.AlarmType_CORRUPT}},
+			healthCheckURL:   "/health?exclude=NOSPACE",
+			expectStatusCode: http.StatusServiceUnavailable,
+			expectHealth:     "false",
+		},
+		{
+			name:             "Unhealthy if both NOSPACE and CORRUPT are on and excluded",
+			alarms:           []*pb.AlarmMember{{MemberID: uint64(0), Alarm: pb.AlarmType_NOSPACE}, {MemberID: uint64(1), Alarm: pb.AlarmType_CORRUPT}},
+			healthCheckURL:   "/health?exclude=NOSPACE&exclude=CORRUPT",
+			expectStatusCode: http.StatusOK,
+			expectHealth:     "true",
+		},
+		{
+			healthCheckURL:   "/health",
+			apiError:         auth.ErrUserEmpty,
+			expectStatusCode: http.StatusOK,
+			expectHealth:     "true",
+		},
+		{
+			healthCheckURL:   "/health",
+			apiError:         auth.ErrPermissionDenied,
+			expectStatusCode: http.StatusOK,
+			expectHealth:     "true",
+		},
+		{
+			healthCheckURL:   "/health",
+			apiError:         fmt.Errorf("Unexpected error"),
+			expectStatusCode: http.StatusServiceUnavailable,
+			expectHealth:     "false",
+		},
+	}
+
+	for i, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mux := http.NewServeMux()
+			HandleHealth(zap.NewExample(), mux, &fakeHealthServer{
+				fakeServer: fakeServer{alarms: tt.alarms},
+				health:     tt.expectHealth,
+				apiError:   tt.apiError,
+			})
+			ts := httptest.NewServer(mux)
+			defer ts.Close()
+
+			res, err := ts.Client().Do(&http.Request{Method: http.MethodGet, URL: testutil.MustNewURL(t, ts.URL+tt.healthCheckURL)})
+			if err != nil {
+				t.Errorf("fail serve http request %s %v in test case #%d", tt.healthCheckURL, err, i+1)
```

Comment:
nit, the tt.name will be printed automatically, so the case number (i+1) is unnecessary.  The same for the following couple of lines. 

It isn't a big deal, feel free to keep it unchanged.

Reply:
Let's clean those up, no reason to leave a mess. Thanks for pointing this out